### PR TITLE
DydxBridge Contract Integration Tests

### PR DESCRIPTION
**Pull request by hysz**
## Description

DydxBridge Contract integration tests with the `ERC20BridgeProxy`, `Exchange` and mainnet `dYdX SoloMargin`.

Included in this PR: 
  1. Added `ERC20BridgeProxy.transferFrom` tests for DydxBridge
  2. Added dydx bridge to `contract-addresses` package
  3. Added integration tests for DydxBridge with Exchange contract
  4. Increased the mocha timeout in the integrations package for mainnet tests.
  5. Added encode/decode of dydxBridgeData to assetDataUtils

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/integrations/test/exchange/fill_dydx_order_test.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        });
+        testContract = await deployDydxBridgeAsync(deployment, env);
+        const encodedBridgeData = dydxBridgeDataEncoder.encode({ bridgeData });
+        const bridgeProxyAssetData = deployment.assetDataEncoder
+            .ERC20Bridge(testContract.address, testContract.address, encodedBridgeData)
+            .getABIEncodedTransactionData();
+        [makerToken, takerToken] = deployment.tokens.erc20;
+
+        // Configure Maker & Taker.
+        const orderConfig = {
```
Comment (mzhu25) [inline contracts/integrations/test/exchange/fill_dydx_order_test.ts]: We should consider adding some tests for dydx bridge assetData as the `takerAssetData`, `makerFeeAssetData`, `takerFeeAssetData` to be safe

Comment (hysz) [reply contracts/integrations/test/exchange/fill_dydx_order_test.ts]: Yeah, this could be good to demonstrate how the bridge would be used in other ways.

File: contracts/asset-proxy/test/dydx_bridge.ts [hunk 2]
```diff
@@ -122,21 +119,50 @@ blockchainTests.resets('DydxBridge unit tests', env => {
...
                     secondaryMarketId: constants.ZERO_AMOUNT,
-                    otherAddress: action.actionType === BridgeActionType.Deposit ? from : to,
+                    otherAddress: action.actionType === DydxBridgeActionType.Deposit ? from : to,
                     otherAccountId: constants.ZERO_AMOUNT,
                     data: '0x',
                 });
             }
             verifyEventsFromLogs(txReceipt.logs, expectedOperateActionEvents, TestDydxBridgeEvents.OperateAction);
         };
+        it('succeeds when calling with zero amount', async () => {
```
[the remark this answers is not in the document]
Comment (hysz) [reply contracts/asset-proxy/test/dydx_bridge.ts]: Yeah that's true. Although I feel like for local unit tests it could be good to include the edge cases, in case we change how the external code works.

File: .circleci/config.yml [hunk 3]
```diff
@@ -23,7 +23,7 @@ jobs:
             #       command: npm set prefix=/home/circleci/npm && echo 'export PATH=$HOME/circleci/npm/bin:$PATH' >> /home/circleci/.bashrc
             - run:
                   name: install-yarn
-                  command: npm install --global yarn@1.17.0
+                  command: npm install --global yarn@1.17.0 || echo "Skipping install"
```
Comment (hysz) [inline .circleci/config.yml]: Temporarily changed to get circle to work. Will not get merged. See issue [here](https://github.com/yarnpkg/yarn/issues/7773).

File: contracts/integrations/test/exchange/fill_dydx_order_test.ts [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+    const dydxConversionRateDenominator = toBaseUnitAmount(1);
+    let maker: Maker;
+    let taker: Taker;
+    let orderConfig: Partial<SignedOrder>;
+    let dydxBridgeProxyAssetData: string;
+    const defaultDepositAction = {
+        actionType: DydxBridgeActionType.Deposit as number,
+        accountId: constants.ZERO_AMOUNT,
+        marketId,
+        conversionRateNumerator: dydxConversionRateNumerator,
```
Comment (abandeali1) [inline contracts/integrations/test/exchange/fill_dydx_order_test.ts]: Can we also test the behavior when the conversion rate would trigger a rounding error?

File: contracts/test-utils/src/web3_wrapper.ts [hunk 5]
```diff
@@ -26,11 +27,15 @@ if (process.env.FORK_RPC_URL !== undefined) {
...
-        // ZeroExGovernor signer addresses
         unlocked_accounts: [
+            // ZeroExGovernor signer addresses
             '0x257619b7155d247e43c8b6d90c8c17278ae481f0',
             '0x5ee2a00f8f01d099451844af7f894f26a57fcbf2',
             '0x894d623e0e0e8ed12c4a73dada999e275684a37d',
+            // Test dYdX user account (can trade DAI on dydx bridge)
+            '0xbd67dce6348dc5949a8af5888d6a2bd5dc3cb86d',
+            // ERC20BridgeProxy
+            getContractAddressesForChainOrThrow(1).erc20BridgeProxy,
```
[the remark this answers is not in the document]
Comment (hysz) [reply contracts/test-utils/src/web3_wrapper.ts]: Figured in the slim chance we do change the value, it could be a headache to debug failed tests 🤷‍♀ .

File: contracts/integrations/test/bridges/dydx_bridge_mainnet_test.ts [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    };
+    const defaultWithdrawAction = {
+        actionType: DydxBridgeActionType.Withdraw as number,
+        accountId: constants.ZERO_AMOUNT,
+        marketId: daiMarketId,
+        // This ratio must be less than the `1` to account
+        // for interest in dydx balances because the test
+        // account has an initial dydx balance of zero.
+        conversionRateNumerator: new BigNumber(1),
+        conversionRateDenominator: new BigNumber(2),
```
[the remark this answers is not in the document]
Comment (hysz) [reply contracts/integrations/test/bridges/dydx_bridge_mainnet_test.ts]: Added to the unit tests. Since execution will fail before calling dydx I don't think we need to run it on the mainnet contract.
